waves/timelock/test/timelock.ride-test.js

```diff
@@ -1,260 +1,249 @@
+    });
+
+    it('should lock the funds', async () => {
+      const timelockTxParams = {
+        dApp: address(accounts.script),
+        call: {
+          function: 'timelock',
+          args: [
+            { type: 'integer', value: totalChunks },
+            { type: 'integer', value: interval },
+            { type: 'integer', value: initialDelay },
+          ],
+        },
+      };
+      const timelock = invokeScript(timelockTxParams, accounts.owner);
+
+      await broadcast(timelock);
+      await waitForTx(timelock.id);
+
+      lockBlock = await height();
+    });
+
+    it('should prevent locking while in a locked state', async () => {
+      const timelockTxParams = {
+        dApp: address(accounts.script),
+        call: {
+          function: 'timelock',
+          args: [
+            { type: 'integer', value: totalChunks },
+            { type: 'integer', value: interval },
+            { type: 'integer', value: initialDelay },
+          ],
+        },
+      };
+      const timelock = invokeScript(timelockTxParams, accounts.owner);
+      expect(broadcast(timelock)).to.be.rejectedWith('Lock already in place');
+    });
+
+    it('should prevent withdrawals until some funds are unlocked', async () => {
+      const withdraw = invokeScript({
+        dApp: address(accounts.script),
+        call: {
+          function: 'withdraw',
+          args: [{ type: 'integer', value: 42 * wvs }],
+        },
+
+      }, accounts.owner);
+
+      expect(broadcast(withdraw)).to.be.rejectedWith('All funds are currently locked');
+    });
+
+    it('should withdraw only unlocked funds', async () => {
+      const timelockTxParams = {
+        dApp: address(accounts.script),
+        call: {
+          function: 'timelock',
+          args: [
+            { type: 'integer', value: totalChunks },
+            { type: 'integer', value: interval },
+            { type: 'integer', value: initialDelay },
+          ],
+        },
+      };
+
+      const withdrawParams = {
+        dApp: address(accounts.script),
+        call: {
+          function: 'withdraw',
+          args: [{ type: 'integer', value: 42 * wvs }],
+        },
+
+      };
+
+      const earlyWithdraw = invokeScript(withdrawParams, accounts.owner);
+      await expect(broadcast(earlyWithdraw)).to.be.rejectedWith('All funds are currently locked');
+
+      await waitUntilBlock(lockBlock + interval + initialDelay);
+
+      const untappedWithdraw = invokeScript(withdrawParams, accounts.owner);
+
+      await broadcast(untappedWithdraw);
+      await waitForTx(untappedWithdraw.id);
+
+      const exhaustedWithdraw = invokeScript(withdrawParams, accounts.owner);
+      await expect(broadcast(exhaustedWithdraw)).to.be.rejectedWith('All funds are currently locked');
+
+      const timelock = invokeScript(timelockTxParams, accounts.owner);
+      expect(broadcast(timelock)).to.be.rejectedWith('Lock already in place');

```

Comment:
A few of the `it` call back functions above have no `expect ` calls so they themselves would't be testing anything. I'm assuming the test involves them just executing without triggering an error or make latter tests work?

Reply:
exactly, the testing suite here is a bit odd since it relies heavily on the execution order to make the changes to the blockchain layer.